node/node.go

```diff
@@ -110,6 +110,8 @@ type Node struct {
 
 	Bwc *metrics.BandwidthCounter
 
+	ArEngine *autoretrieve.AutoretrieveEngine
```

Comment:
from my comments above, there will be no need to have this in the node

Reply:
Same as above, we need to access the `s` (server) to query the database